File: a_easy/07_simple_symbols/simple_symbols.rb
def SimpleSymbols(str)
  bools = []

  for n in 0..str.length-1 do
    if str[n].match(/[A-Za-z]/)
      if n == 0
        bools << false
      elsif n == str.length-1
        bools << (str[n-1] == '+')
      else
        bools << (str[n-1] == '+' && str[n+1] == '+')
      end
    end
  end

  bools.all?
end

# keep this function call here
puts SimpleSymbols(STDIN.gets)  

File: a_easy/07_simple_symbols/simple_symbols.js
function simpleSymbols(str) {
  var bools = [];
  var array = str.split("");

  for (i = 0; i < str.length; i++) {
    if (/[A-Za-z]/.test(array[i]) && i !== 0 && i !== array.length) {
      if (array[i - 1] === "+" && array[i + 1] === "+") {
        bools.push(true);
      } else {
        bools.push(false);
      }
    }
  }

  return !bools.includes(false)
}
